File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        
        if (s.length() != t.length()) {
            return false;
        }
        
        int[] arr = new int[26];
        char[] sArr = s.toCharArray();
        char[] tArr = t.toCharArray();
        
        for(int i=0; i < s.length(); i++) {
            arr[sArr[i] - 'a']++;
            arr[tArr[i] - 'a']--;
        }
        
        for(int i=0; i < 26; i++) {
            if (arr[i] != 0) {
                return false;
            }
        }
        
        return true;
        
    }
}

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        
        if (s.size() != t.size()) {
            return false;
        }
        
        
        unordered_map<char, int> map;
        
        for(char c:s) {
            map[c]++;
        }
        
        for(char c:t){
            if(map.find(c) == map.end() || map[c] == 0) {
                return false;
            }    
            else {
                map[c]--;
            }
        }
        return true;
        
        
    }
};